From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Thu, 25 Nov 2010 09:53:47 +0100
Subject: [PATCH] add support for LDFLAGS, don't overwrite CFLAGS

When cross compiling the linker can not find libz, this patch
adds the LDFLAGS variable to pass the linkerflags.

Based on earier work by Erwin Rol.

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 squashfs-tools/Makefile |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/squashfs-tools/Makefile b/squashfs-tools/Makefile
index 183578b..8d2fa35 100644
--- a/squashfs-tools/Makefile
+++ b/squashfs-tools/Makefile
@@ -87,7 +87,7 @@ MKSQUASHFS_OBJS = mksquashfs.o read_fs.o sort.o swap.o pseudo.o compressor.o
 UNSQUASHFS_OBJS = unsquashfs.o unsquash-1.o unsquash-2.o unsquash-3.o \
 	unsquash-4.o swap.o compressor.o
 
-CFLAGS = $(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
+CFLAGS += $(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-D_GNU_SOURCE -DCOMP_DEFAULT=\"$(COMP_DEFAULT)\"  -O2 -Wall
 
 LIBS =
@@ -175,7 +175,7 @@ endif
 all: mksquashfs unsquashfs
 
 mksquashfs: $(MKSQUASHFS_OBJS)
-	$(CC) $(MKSQUASHFS_OBJS) -lpthread -lm $(LIBS) -o $@
+	$(CC) $(MKSQUASHFS_OBJS) -lpthread -lm $(LIBS) $(LDFLAGS) -o $@
 
 mksquashfs.o: mksquashfs.c squashfs_fs.h mksquashfs.h global.h sort.h \
 	squashfs_swap.h xattr.h
@@ -195,7 +195,7 @@ xattr.o: xattr.h
 read_xattrs.o: xattr.h
 
 unsquashfs: $(UNSQUASHFS_OBJS)
-	$(CC) $(UNSQUASHFS_OBJS) -lpthread -lm $(LIBS) -o $@
+	$(CC) $(UNSQUASHFS_OBJS) -lpthread -lm $(LIBS) $(LDFLAGS) -o $@
 
 unsquashfs.o: unsquashfs.h unsquashfs.c squashfs_fs.h squashfs_swap.h \
 	squashfs_compat.h global.h xattr.h
-- 
1.7.2.3

